app/engine/Condition.scala

```diff
@@ -41,25 +40,25 @@ trait Condition {
 
   val auditType: Option[RoutingReason]
 
-  def isTrue(authContext: AuthContext, ruleContext: RuleContext)(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Boolean]
+  def isTrue(ruleContext: RuleContext)(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Boolean]
 
-  def evaluate(authContext: AuthContext, ruleContext: RuleContext, auditContext: TAuditContext)(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Boolean] = {
-    this.isTrue(authContext, ruleContext).andThen { case Success(result) if auditType.isDefined => auditContext.setRoutingReason(auditType.get, result) }
+  def evaluate(ruleContext: RuleContext, auditContext: TAuditContext)(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Boolean] = {
+    this.isTrue(ruleContext).andThen { case Success(result) if auditType.isDefined => auditContext.setRoutingReason(auditType.get, result) }
   }
 
   def and(other: Condition): Condition = new CompositeCondition {
 
-    override def evaluate(authContext: AuthContext, ruleContext: RuleContext, auditContext: TAuditContext)(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Boolean] = {
-      val condition1FutureResult: Future[Boolean] = self.evaluate(authContext, ruleContext, auditContext)
-      condition1FutureResult.flatMap(c1r => if (c1r) other.evaluate(authContext, ruleContext, auditContext).map(c2r => c1r && c2r) else condition1FutureResult)
+    override def evaluate(ruleContext: RuleContext, auditContext: TAuditContext)(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Boolean] = {
+      val condition1FutureResult: Future[Boolean] = self.evaluate(ruleContext, auditContext)
+      condition1FutureResult.flatMap(c1r => if (c1r) other.evaluate(ruleContext, auditContext).map(c2r => c1r && c2r) else condition1FutureResult)
     }
   }
 
   def or(other: Condition): Condition = new CompositeCondition {
 
-    override def evaluate(authContext: AuthContext, ruleContext: RuleContext, auditContext: TAuditContext)(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Boolean] = {
-      val condition1FutureResult: Future[Boolean] = self.evaluate(authContext, ruleContext, auditContext)
-      condition1FutureResult.flatMap(c1r => if (c1r) condition1FutureResult else other.evaluate(authContext, ruleContext, auditContext))
+    override def evaluate(ruleContext: RuleContext, auditContext: TAuditContext)(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Boolean] = {
+      val condition1FutureResult: Future[Boolean] = self.evaluate(ruleContext, auditContext)
```

Comment:
Can you give `condition1FutureResult` a better name please. And the :type shouldn't be needed